errors.go

```diff
@@ -297,6 +308,62 @@ func FromGRPCStatus(grpcStatus status.Status) *AsertoError {
 	return result
 }
 
+/**
+ * Retrieves the most inner logger associated with an error.
+ */
+func Logger(err error) *zerolog.Logger {
+	var logger *zerolog.Logger
+
+	if err == nil {
+		return logger
+	}
+
+	for {
+		wErr, ok := err.(*werr.WrappedError)
+		if ok {
+			aErr, aOk := wErr.Err.(*AsertoError)
+			if aOk {
+				setLogger(aErr.Ctx, &logger)
+			}
+			setLogger(wErr.Ctx, &logger)
+		}
+
+		aErr, ok := err.(*AsertoError)
+		if ok {
+			setLogger(aErr.Ctx, &logger)
+
+		}
+
+		err = errors.Unwrap(err)
+		if err == nil {
+			break
+		}
+	}
+
+	return logger
+}
+
+/**
+ * setLogger sets the logger pointer to the logger stored in the provided context.
+ * If the context is nil or the logger in the context is nil, the logger pointer remains unchanged.
+ * If the logger in the context is the default context logger or has a disabled level, the logger pointer remains unchanged.
+ *
+ * @param ctx The context from which to retrieve the logger.
+ * @param logger The pointer to the logger to be set.
+ */
+func setLogger(ctx context.Context, logger **zerolog.Logger) {
```

Comment:
Pointers to pointers are highly unusual and can easily be error prone.
I'd write this function as
```go
func extractLogger(ctx context.Context) *zerolog.Logger {
...
}
```

Reply:
That was my initial implementation, cool, i'll revert to that